### DISCUSSION TITLE
Comments under Checkins

### DISCUSSION THREAD

**Discussion by Oxbowatos** [Traewelling/traewelling#2035] [Ideas / Feature Requests]
General Idea is that peoples can write comments under checkins.

In the Settings you can set who can comment under your checkins like: People who Follow you or People who you follow or Public all that are logged in.

The Comments should be deletable by the commenter and checkin owner and editable by the commenter.

The Checkin Owner also gets a Notification when someone comments on their checkin.

The Comments can look like this:
![grafik](https://user-images.githubusercontent.com/123954576/216022734-504a2d15-8012-47ec-8298-d36c5ae4ca7e.png)

**Comment by MrKrisKrisu** [comment]
I would also be happy about such a feature. However, we should first think about the moderation of the content.

**Comment by Oxbowatos** [comment]
Content Moderation could be easy.
You could make an Automod system thats client based which already checking for banned words before sending it or a backend way to check for banned words.

And maybe you could make a system like on YouTube, when a comment has a banned word it gets under a review mode where the Checkin Owner can accept the comment or delete it before it gets public.

**Comment by NyCodeGHG** [comment]
It could also be possible to integrate https://perspectiveapi.com/ to automatically flag in addition to manual content moderation.

**Reply by MrKrisKrisu** [reply]
Do you know if there is some similar (self hosted) project without exposing data to external services?

**Reply by NyCodeGHG** [reply]
https://github.com/IBM/MAX-Toxic-Comment-Classifier this looks good, but from what i tried, it does not seem to support the german language.

**Comment by ChristianTacke** [comment]
Is Träweeling slowly becoming a mini-mastodon?

Then it should also support federation! :wink:
Start by supporting [WebFinger](https://docs.joinmastodon.org/spec/webfinger/)?

**Comment by MrKrisKrisu** [comment]
> Is Träweeling slowly becoming a mini-mastodon?
> 
> Then it should also support federation! 😉 Start by supporting [WebFinger](https://docs.joinmastodon.org/spec/webfinger/)?

maybe move this part of discussion to #932? Sounds similar.

**Comment by NyCodeGHG** [comment]
For WebFinger see #2153
